Q3 Hackathon: Tooltip Shortcode

brett0000FF opened the pull request:
<!-- *Note: Please remember to review the Datadog Documentation [Contribution Guidelines](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md) if you have not yet done so.* -->

### What does this PR do? What is the motivation?
<!-- A brief description of the change being made with this pull request. What is your motivation for the PR? -->
This new shortcode enhances our documentation by allowing authors to add interactive, informative tooltips to terms or phrases. It improves readability by providing additional context without cluttering the main text.

The shortcode supports two main use cases:

- Custom tooltip content
- Reuse glossary content

#### Custom tooltip

Provide a `text` and `tooltip` param to include custom tooltip content on the page.

* `text`: The underlined text to display on the page.
* `tooltip`: The tooltip text that pops in on hover.

For example:

```markdown
Here's an example of a {{< tooltip text="tooltip" tooltip="This is additional information that appears in the tooltip" >}} in action.
```

https://github.com/user-attachments/assets/78fe83b5-03c5-401e-b6ce-63888f10b3bc

#### Glossary tooltip

Provide a `glossary` param that corresponds with a term in the glossary. Use the term as the pre-hover text. If `short_definition` exists in the glossary frontmatter, show that. When short definition is shown, a **Glossary** link is provided in the tooltip to the full definition.

* `glossary`: The glossary term to pull content from and display.
* (frontmatter) `short_definition`: The condensed definition to display if available.

For example, using full definition:

```markdown
Here's an example of a {{< tooltip glossary="archive" >}} in action.
```

https://github.com/user-attachments/assets/cebefa97-095c-414a-817c-fb7d2cbc9f42

For example, using short definition:

```markdown
---
title: trace root span
core_product:
  - apm
short_definition: 'A span is a trace root span when it is the first span of a trace. The root span is the entry-point method of the traced request. Its start marks the beginning of the trace.'
---
```

```markdown
Here's an example of a {{< tooltip glossary="trace root span" >}} in action.
```

https://github.com/user-attachments/assets/ceda3981-1f29-468a-890b-b3c6f6e813ea

#### Case options

Use `case` to force the tooltip text to a specific case. This might be necessary if the tooltip is the first word in a sentence.
- **Default**: Matches case of glossary term title.
- `title`: Term is Title Case
- `lower`: term is lower case
- `upper` TERM IS UPPER CASE
- `sentence` Term is sentence case

### Merge instructions
<!-- If you want us to merge this PR as soon as we've reviewed, check the box below. If you're waiting for a release or there are other considerations that you want us to be aware of, list them below. -->

- [ ] Please merge after reviewing

### Additional notes
<!-- Anything else we should know when reviewing?-->

<!-- Previewing the PR: Assuming you are a Datadog employee and named your branch `<yourname>/<description>`, a preview build will run and links to the preview output will be auto-generated and posted in the PR comments. The links will 404 until the preview build is finished running. -->

bgdeutsch reviewed (changes requested):
Hey @brett0000FF, code changes look good here!   Can you generate a preview with some examples so we can do some QA?   Thanks!

brett0000FF commented:
> * https://docs-staging.datadoghq.com/brett0000FF/hackathon-hover/test

@bgdeutsch - Thanks for taking a look! I added [a test page](https://docs-staging.datadoghq.com/brett0000FF/hackathon-hover/test) with these examples.

Two quick notes:
- I forgot that the display text would be forced to english when using the `glossary` option. I just changed display text to be .Title, so it will match the translated term now.
- Locally, everything looks fine, but in the staging preview, I see that it's ignoring the space at the end of the shortcode. Do you have any idea why?

Thanks!

StefonSimmons commented on layouts/shortcodes/tooltip.html:
```diff
@@ -0,0 +1,47 @@
+<!-- Render the tooltip -->
+<span class="tooltip-container" x-data="{ open: false }" @mouseenter="open = true" @mouseleave="open = false">
+  <button class="tooltip-trigger">{{ $displayText }}</button>
+  <span id="tooltip-{{ $displayText | urlize }}" class="tooltip-content" x-bind:class="{ 'show': open }" role="tooltip">
```
```suggestion
  <span id="tooltip-{{ $displayText | anchorize }}" class="tooltip-content" x-bind:class="{ 'show': open }" role="tooltip">
```

@brett0000FF I recommend using `anchorize` instead of `urlize` here https://gohugo.io/functions/urls/anchorize/#readout

brett0000FF commented on assets/styles/components/_tooltip.scss:
```diff
@@ -0,0 +1,110 @@
+
+.tooltip-container {
+  position: relative;
+  display: inline-block;
```
@StefonSimmons - This seems to force a space after the shortcode. For example, imagine a scenario where we want a period directly after the tooltip:

![image](https://github.com/user-attachments/assets/8f0f4c3a-2bac-479e-8042-6e1e2093a0f1)

I pushed up a change where I revert inline-block and keep anchorize to see how it looks on Staging. This combo works locally.

brett0000FF replied:
Ok, when I revert inline block, it's back to not allowing any spaces. I'm going to put it back to inline because that is more important than being able to use this at the end of a sentence for now.

jhgilbert commented:
The tooltip works nicely on mobile when I tap it, but there's no styling to indicate I should tap it. Do we want to ensure that the dotted underline shows up on mobile? I realize mobile isn't our primary concern but still nice to offer support if it's easy to do.

![IMG_1364](https://github.com/user-attachments/assets/2d621b07-62fc-49fc-b054-c58eacb9226e)

![IMG_1365](https://github.com/user-attachments/assets/73bf064a-687c-4e71-900e-6fedfc76ef23)

I'm not sure how extensively we're expected to browser-test new features like this, but just wanted to raise it, in case we should be testing it outside of Chrome etc. Normally I'm not so paranoid about it, but tooltips in particular can be a little wonky sometimes. Here's [the list of browsers](https://datadoghq.atlassian.net/wiki/spaces/FRON/pages/2639497894/Supported+browsers+for+Datadog+web+app#Currently-supported-browsers) officially supported by Datadog, but I defer to websites on how much of this is necessary.

I'll do a more in-depth review once websites has finished looking at this -- and of course let me know if I can help!

brett0000FF commented:
> The tooltip works nicely on mobile when I tap it, but there's no styling to indicate I should tap it. Do we want to ensure that the dotted underline shows up on mobile? I realize mobile isn't our primary concern but still nice to offer support if it's easy
 
@jhgilbert - Good catch! Curious what Websites thinks too. To fix this, I think we can remove all `text-decoration` from the CSS and replace with something like `border-bottom: 1px dotted currentColor;`. I'll wait to change anything for now though. 👍
